src/smt/proof_manager.h

```diff
@@ -143,6 +144,12 @@ class PfManager : protected EnvObj
       std::shared_ptr<ProofNode> pfn,
       Assertions& as,
       ProofScopeMode scopeMode = ProofScopeMode::UNIFIED);
+  /**
+   * Check proof. This call runs the final proof callback, which checks for
+   * pedantic failures and takes statistics.
+   * @param pfn The proof to check.
+   */
+  void checkFinalProof(std::shared_ptr<ProofNode> pfn);
```

Comment:
It's not really accurate to say that this checks the proof, right? In the regular sense of "checking" the proof is checked by being built, which is not done by this method. Maybe we could call this something else? "checkPedanticAndCollectStats" maybe.

Reply:
It does check if e.g. `--proof-check=lazy`.